controllers/componentplan_controller.go

```diff
@@ -143,29 +143,46 @@ func (r *ComponentPlanReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	isPlanMarkedToBeDeleted := plan.GetDeletionTimestamp() != nil
 	if isPlanMarkedToBeDeleted && controllerutil.ContainsFinalizer(plan, corev1alpha1.Finalizer) {
 		logger.Info("Performing Finalizer Operations for Plan before delete CR")
-
-		// Perform all operations required before remove the finalizer and allow
-		// the Kubernetes API to remove ComponentPlan
-		if err = r.doFinalizerOperationsForPlan(ctx, logger, getter, plan); err != nil {
-			logger.Error(err, "Failed to re-fetch ComponentPlan")
-			return ctrl.Result{}, err
+		if plan.InActionedReason(corev1alpha1.ComponentPlanReasonUnInstalling) {
+			logger.Info("the last one is uninstalling... skip for 1 minute")
+			return ctrl.Result{RequeueAfter: time.Minute}, nil
 		}
-
-		logger.Info("Removing Finalizer for ComponentPlan after successfully performing the operations")
-		if ok := controllerutil.RemoveFinalizer(plan, corev1alpha1.Finalizer); !ok {
-			return ctrl.Result{Requeue: true, RequeueAfter: 3 * time.Second}, nil
+		if plan.InActionedReason(corev1alpha1.ComponentPlanReasonUnInstallFailed) {
+			// TODO do we need uninstall retry like install or upgrade done?
+			logger.Info("the last one is uninstallfailed, just skip this one")
```

Comment:
good point! we can do `uninstall retry` here but with maybe only `one more time???`

Reply:
will track in #120